packages/circuits/test/epochKeyLite.test.ts

```diff
@@ -0,0 +1,190 @@
+        for (let nonce = 0; nonce < NUM_EPOCH_KEY_NONCE_PER_EPOCH; nonce++) {
+            const attesterId = 10210
+            const epoch = 120958
+            const id = new ZkIdentity()
+            const circuitInputs = {
+                identity_nullifier: id.identityNullifier,
+                control: EpochKeyLiteProof.buildControlInput({
+                    epoch,
+                    attesterId,
+                    nonce,
+                }),
+                data: 0,
+            }
+            const { isValid, publicSignals, proof } = await genProofAndVerify(
+                Circuit.epochKeyLite,
+                circuitInputs
+            )
+            expect(isValid).to.be.true
+            const data = new EpochKeyLiteProof(publicSignals, proof)
+            expect(data.epochKey.toString()).to.equal(
+                genEpochKey(
+                    id.identityNullifier,
+                    attesterId,
+                    epoch,
+                    nonce
+                ).toString()
+            )
+            expect(data.control).to.equal(
+                (
+                    (BigInt(attesterId) << BigInt(72)) +
+                    (BigInt(epoch) << BigInt(8))
+                ).toString()
+            )
+            expect(data.epoch.toString()).to.equal(epoch.toString())
+            expect(data.nonce.toString()).to.equal('0')
+            expect(data.revealNonce.toString()).to.equal('0')
+            expect(data.attesterId.toString()).to.equal(attesterId.toString())
+            expect(data.data.toString()).to.equal('0')
+        }
+    })
+
+    it('should reveal nonce', async () => {
+        for (let nonce = 0; nonce < NUM_EPOCH_KEY_NONCE_PER_EPOCH; nonce++) {
+            const attesterId = 10210
+            const epoch = 120958
+            const id = new ZkIdentity()
+            const circuitInputs = {
+                identity_nullifier: id.identityNullifier,
+                control: EpochKeyLiteProof.buildControlInput({
+                    epoch,
+                    attesterId,
+                    nonce,
+                    revealNonce: 1,
+                }),
+                data: 0,
+            }
+            const { isValid, publicSignals, proof } = await genProofAndVerify(
+                Circuit.epochKeyLite,
+                circuitInputs
+            )
+            expect(isValid).to.be.true
+            const data = new EpochKeyLiteProof(publicSignals, proof)
+            expect(data.epochKey.toString()).to.equal(
+                genEpochKey(
+                    id.identityNullifier,
+                    attesterId,
+                    epoch,
+                    nonce
+                ).toString()
+            )
+            expect(data.control).to.equal(
+                (
+                    (BigInt(1) << BigInt(232)) +
+                    (BigInt(attesterId) << BigInt(72)) +
+                    (BigInt(epoch) << BigInt(8)) +
+                    BigInt(nonce)
```

Comment:
can the shift bits be more readable?
e.g.
```ts
(BigInt(attesterId) << BigInt( EPOCH_BITS + NONCE_BITS )) +
BigInt(epoch) << BigInt(NONCE_BITS))
```
or it can be concatenated
```ts
const data = revealNonce +
             attesterId.padStart(ATTESTER_ID_BITS) + 
             epoch.padStart(EPOCH_BITS) + ...
```

Reply:
Yea using constants defined in the proof class would make this more clear.

Reply:
`padStart` shouldn't work, it has to be a numerical addition

Reply:
This is out of scope for this pull request, documented in #238